ext/intl/dateformat/dateformat_create.cpp

```diff
@@ -99,7 +99,11 @@ static zend_result datefmt_ctor(INTERNAL_FUNCTION_PARAMETERS, zend_error_handlin
 	}
 	if (!INTL_UDATE_FMT_OK(time_type)) {
 		intl_error_set(NULL, U_ILLEGAL_ARGUMENT_ERROR, "datefmt_create: invalid time format style", 0);
-return FAILURE;
+		return FAILURE;
+	}
+	if (date_type == UDAT_PATTERN && time_type != UDAT_PATTERN) {
+		intl_error_set(NULL, U_ILLEGAL_ARGUMENT_ERROR, "datefmt_create: invalid time format style, UDAT_PATTERN expected", 0);
```

Comment:
The error message is confusing here, I think something along the usual line of ``time format must be UDAT_PATTERN if date format is UDAT_PATTERN`` is clearer.

May or may not make sense to convert this to a ValueError in master